pkg/flow/events.go

```diff
@@ -47,6 +50,13 @@ type events struct {
 	*server
 }
 
+type keyPair struct {
+	namespace  string
+	filtername string
+}
+
+var cache = make(map[keyPair]string)
```

Comment:
This cache is not safe. Concurrent map access where at least one operation is not read-only can cause extremely rare and difficult-to-reproduce panics. Since all changes to this mpa happen in gRPC handlers, they're completely asynchronous, which means concurrent access can definitely happen. To be safe, you need to use `sync.Map` or mutexes. 

However, I think for now it's best to remove the cache entirely. We will soon be moving to Redis to handle caching. Very soon. So whatever caching we build/fix here is not going to stay in the codebase for very long.

Reply:
This should be fine. I've unresolved the conversation for a small adjustment: the variable name `cache` is too generic here. This code exists within a very large package called `flow`, and although it's currently the only cache, the size of the package means it should be viewed almost like a global variable (even though it's unexported). I can also see it being renamed as soon as the redis stuff comes. How about we change it to `eventFilterCache`?